libkbfs/journal_md_ops.go

```diff
@@ -91,6 +91,26 @@ func (j journalMDOps) getHeadFromJournal(
 		return ImmutableRootMetadata{}, nil
 	}
 
+	// If we are being asked for the merged head, but the journal is
```

Comment:
I'm beginning to think this filtering logic is better put in `mdJournal.getHead()`. You can change it to take a `bid BranchID` have it do the filtering, and have it return `ImmutableBareRootMetadata{}, nil` if the head's BID doesn't match. It can do the filtering quickly without loading the MD; the only special case is if `bid == NullBranchID` and `mdJournal.branchID == PendingLocal...`, in which case you can iterate through the md ID journal looking for the latest one with `IsLocalSquash` set (if any).